### PR TITLE
[DynamoDb] Batch write item

### LINKED ISSUE #566: [DynamoDb] Generate batchWriteItem() and batchGetItem()
This might need some more logic in the code generator. 

It is needed for #551

### PR DESCRIPTION
**Pull request by Nyholm**
This will fix #566

~~Blocked by #677~~

### CODE REVIEW

File: src/Service/DynamoDb/src/DynamoDbClient.php [hunk 1]
```diff
@@ -35,6 +37,28 @@
...
 {
+    /**
+     * The `BatchWriteItem` operation puts or deletes multiple items in one or more tables. A single call to
+     * `BatchWriteItem` can write up to 16 MB of data, which can comprise as many as 25 put or delete requests. Individual
+     * items to be written can be as large as 400 KB.
+     *
+     * @see https://docs.aws.amazon.com/aws-sdk-php/v3/api/api-dynamodb-2012-08-10.html#batchwriteitem
+     *
+     * @param array{
+     *   RequestItems: array<string, array>,
```
Comment (Nyholm) [inline src/Service/DynamoDb/src/DynamoDbClient.php:48]: This is not super pretty. But Im okey with this.

File: src/CodeGenerator/src/Generator/InputGenerator.php [hunk 2]
```diff
@@ -154,8 +154,25 @@ public function generate(Operation $operation): ClassName
...
+                        throw new \RuntimeException('Recursive ListShape with non StructureShape member is not implemented.');
+                    }
+                    $memberClassName = $this->objectGenerator->generate($listMemberShape);
+                    $constructorBody .= strtr('
+                        $this->NAME = [];
+                        foreach ($input["NAME"] ?? [] as $key => $item) {
+                            $this->NAME[$key] = array_map(function($v) {return CLASS::create($v);}, $item);
+                        }
+                    ', [
+                        'NAME' => $member->getName(),
```
Comment (jderusse) [inline src/CodeGenerator/src/Generator/InputGenerator.php:170]: `getLocation() ?? getName()`?

File: src/CodeGenerator/src/Generator/InputGenerator.php [hunk 3]
```diff
@@ -154,8 +154,25 @@ public function generate(Operation $operation): ClassName
...
+                } elseif ($mapValueShape instanceof ListShape) {
+                    $listMember = $mapValueShape->getMember();
+                    $listMemberShape = $listMember->getShape();
+                    if (!$listMemberShape instanceof StructureShape) {
+                        throw new \RuntimeException('Recursive ListShape with non StructureShape member is not implemented.');
+                    }
+                    $memberClassName = $this->objectGenerator->generate($listMemberShape);
+                    $constructorBody .= strtr('
+                        $this->NAME = [];
+                        foreach ($input["NAME"] ?? [] as $key => $item) {
```
Comment (jderusse) [inline src/CodeGenerator/src/Generator/InputGenerator.php:166]: are you sure about $key in a list ?

File: src/Service/DynamoDb/src/DynamoDbClient.php (hunk 1 above)
Comment (jderusse) [reply src/Service/DynamoDb/src/DynamoDbClient.php:70]: should be fixed by https://github.com/async-aws/aws/pull/681

File: src/CodeGenerator/src/Generator/InputGenerator.php (hunk 3 above)
Comment (Nyholm) [reply src/CodeGenerator/src/Generator/InputGenerator.php:166]: Yes. In `BatchWriteItemInput` they specify the database table. 

https://github.com/async-aws/aws/pull/679/files#diff-c63388a34e43ac3b149f15b4b3c58cdcR20

File: src/CodeGenerator/src/Generator/InputGenerator.php (hunk 2 above)
Comment (Nyholm) [reply src/CodeGenerator/src/Generator/InputGenerator.php:170]: We don't do that for the section just above the added code. Im not sure why..

File: src/CodeGenerator/src/Generator/InputGenerator.php (hunk 3 above)
Comment (jderusse) [reply src/CodeGenerator/src/Generator/InputGenerator.php:166]: oh nvm... `foreach ($input["NAME"] ?? [] as $key => $item) {` is iterating over the Map. 
The iteration over the list is done next line with array_map.

Make sens

File: src/CodeGenerator/src/Generator/InputGenerator.php (hunk 2 above)
Comment (jderusse) [reply src/CodeGenerator/src/Generator/InputGenerator.php:170]: maybe because it's a map... I'm fine with this

**Pull request merged**